gefserver/server/events.go

```diff
@@ -10,12 +10,19 @@ import (
 	"strings"
 	"sync/atomic"
 	"time"
+
+	"github.com/EUDAT-GEF/GEF/gefserver/db"
 )
 
-func decorate(fn func(http.ResponseWriter, *http.Request), actionType string) func(http.ResponseWriter, *http.Request) {
+func (s *Server) decorate(fn func(http.ResponseWriter, *http.Request), actionType string) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 		logRequest(r)
-		allow, closefn := signalEvent(actionType, r)
+		user, err := s.getCurrentUser(r)
+		if err != nil {
+			Response{w}.ServerError("User error", err)
```

Comment:
No, writing to the response is not a good idea unless you make sure the response is closed afterwards. We should just log the error.